### PR TITLE
♻️ Refactor backend functions to accept queries

### PR DESCRIPTION
**Pull request by arassaria**
# Description

Refactoring existing backend functions to accept queries instead of long URLs

## Type of change

- [x] Refactoring code

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -50,8 +50,9 @@ app.post("/api/:collectionName", async (req, res) => {
   }
 });
 
-app.delete("/api/:collectionName/:data", async (req, res) => {
-  const { collectionName, data } = req.params;
+app.delete("/api/:collectionName", async (req, res) => {
+  let data = req.query.data;
+  const { collectionName } = req.params;
```
Comment (phlgr) [inline server.js]:
```suggestion
  const { data } = req.query;
  const { collectionName } = req.params;
```

File: server.js [hunk 2]
```diff
@@ -63,8 +64,9 @@ app.delete("/api/:collectionName/:data", async (req, res) => {
   }
 });
 
-app.patch("/api/:collectionName/:dataName", async (req, res) => {
-  const { collectionName, dataName } = req.params;
+app.patch("/api/:collectionName/", async (req, res) => {
+  let dataName = req.query.data;
```
Comment (phlgr) [inline server.js]:
```suggestion
  const dataName = req.query.data;
```

**Review by phlgr — CHANGES_REQUESTED** [review]
Why do these have to be `let`'s

File: server.js (hunk 1 above)
Comment (arassaria) [reply server.js]: Thanks, didn't think of that way 😄

File: server.js (hunk 2 above)
Comment (arassaria) [reply server.js]: Changed it 😄

**Review by bwnstck — APPROVED** [review]
let us be const! 🤘

**Pull request merged**